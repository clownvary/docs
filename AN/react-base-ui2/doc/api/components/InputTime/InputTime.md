## InputTime  extends InputMoment
---
InputTime Component

### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
showTrigger2|bool|true|false|Icon class name for the 2nd trigger button.
format|string|'t'|false|The format pattern to display the date value. {@link InputMoment.InputMomentPropTypes.format}