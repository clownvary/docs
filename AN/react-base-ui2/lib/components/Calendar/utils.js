'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _moment = require('moment');var _moment2 = _interopRequireDefault(_moment);
var _isArray = require('lodash/isArray');var _isArray2 = _interopRequireDefault(_isArray);
var _uniqWith = require('lodash/uniqWith');var _uniqWith2 = _interopRequireDefault(_uniqWith);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

var getTenYearRange = function getTenYearRange(mmomentDate) {
  var year = mmomentDate.year();
  var begin = year - year % 10;
  var end = begin + 9;
  return begin + '-' + end;
};

var getWeeks = function getWeeks(currentDate, firstDayOfWeek) {
  var now = currentDate.clone();
  var veryShortWeekdays = [];
  var weekDays = [];
  var localeData = currentDate.localeData();
  for (var dateColIndex = 0; dateColIndex < 7; dateColIndex += 1) {
    var index = (firstDayOfWeek + dateColIndex) % 7;
    now.day(index);
    weekDays[dateColIndex] = localeData.weekdays(now);
    veryShortWeekdays[dateColIndex] = weekDays[dateColIndex].substring(0, 1);
  }

  return { veryShortWeekdays: veryShortWeekdays, weekDays: weekDays };
};

var compareByFormat = function compareByFormat(date1, date2, format) {return (
    date1.format(format) === date2.format(format));};

var getSafeValue = function getSafeValue(value) {
  if (!(0, _isArray2.default)(value)) {
    value = [value];
  }
  var safeValue = value.map(function (v) {return v && (0, _moment2.default)(v);}).filter(function (v) {return v && v.isValid();});

  return (0, _uniqWith2.default)(safeValue, function (m1, m2) {return m1.isSame(m2, 'day');});
};

var sortedDates = function sortedDates(value) {return value.map(function (d) {return d.clone();}).
  sort(function (a, b) {
    if (a.isAfter(b, 'day')) {
      return 1;
    }
    if (a.isBefore(b, 'day')) {
      return -1;
    }
    return 0;
  });};exports.default =

{
  getTenYearRange: getTenYearRange,
  getWeeks: getWeeks,
  compareByFormat: compareByFormat,
  getSafeValue: getSafeValue,
  sortedDates: sortedDates };module.exports = exports['default'];