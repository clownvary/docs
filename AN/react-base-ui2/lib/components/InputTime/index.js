'use strict';Object.defineProperty(exports, "__esModule", { value: true });exports.InputTimeProps = exports.InputTime = undefined;var _InputTime = require('./InputTime');var _InputTime2 = _interopRequireDefault(_InputTime);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.


InputTime = _InputTime2.default;exports.
InputTimeProps = _InputTime.InputTimeProps;exports.default =


_InputTime2.default;