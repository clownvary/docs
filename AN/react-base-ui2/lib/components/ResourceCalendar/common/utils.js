'use strict';Object.defineProperty(exports, "__esModule", { value: true });exports.getDateText = exports.toDate = exports.getDateKey = exports.DATE_KEY_FORMAT = exports.getEndOfDay = exports.dateIntersect = exports.getDaysInMonth = undefined;var _moment = require('moment');var _moment2 = _interopRequireDefault(_moment);
var _times = require('lodash/times');var _times2 = _interopRequireDefault(_times);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

var getDaysInMonth = exports.getDaysInMonth = function getDaysInMonth(date) {
  var daysInMonth = [];

  var monthDate = (0, _moment2.default)(date).startOf('month');
  (0, _times2.default)(monthDate.daysInMonth(), function () {
    daysInMonth.push(monthDate);
    monthDate = monthDate.clone().add(1, 'd');
  });

  return daysInMonth;
};

var dateIntersect = exports.dateIntersect = function dateIntersect(date, start, end) {
  var dateStart = (0, _moment2.default)(date).startOf('day');
  var dateEnd = (0, _moment2.default)(date).endOf('day');

  if (start.isSameOrBefore(dateEnd) && end.isSameOrAfter(dateStart)) {
    return {
      start: _moment2.default.max(dateStart, start).clone(),
      end: _moment2.default.min(dateEnd, end).clone() };

  }

  return null;
};

var getEndOfDay = exports.getEndOfDay = function getEndOfDay(date, exclusiveMode) {
  var end = date.clone().endOf('day');
  if (exclusiveMode) {
    return end.add(1, 'ms');
  }
  return end;
};

var DATE_KEY_FORMAT = exports.DATE_KEY_FORMAT = 'YYYY-MM-DD';
var getDateKey = exports.getDateKey = function getDateKey(date) {return date.format(DATE_KEY_FORMAT);};
var toDate = exports.toDate = function toDate(key) {return (0, _moment2.default)(key, DATE_KEY_FORMAT);};

var getDateText = exports.getDateText = function getDateText(date) {var format = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'DD ddd';return date.format(format);};