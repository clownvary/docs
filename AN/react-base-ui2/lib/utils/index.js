'use strict';Object.defineProperty(exports, "__esModule", { value: true });exports.shallowEqual = exports.browser = exports.filterValidCustomProps = exports.listenKeyDown = exports.filterBy = exports.cls = exports.momentHelper = exports.decodeHtmlStr = exports.reducerHandler = exports.DataAccess = exports.normalizeKeys = exports.createReducer = exports.createFSA = exports.isFSA = exports.isPromise = exports.toQueryString = undefined;var _toQueryString = require('./toQueryString');var _toQueryString2 = _interopRequireDefault(_toQueryString);
var _isPromise = require('./isPromise');var _isPromise2 = _interopRequireDefault(_isPromise);
var _isFSA = require('./isFSA');var _isFSA2 = _interopRequireDefault(_isFSA);
var _createFSA = require('./createFSA');var _createFSA2 = _interopRequireDefault(_createFSA);
var _createReducer = require('./createReducer');var _createReducer2 = _interopRequireDefault(_createReducer);
var _normalizeKeys = require('./normalizeKeys');var _normalizeKeys2 = _interopRequireDefault(_normalizeKeys);
var _dataAccess = require('./dataAccess');var DataAccess = _interopRequireWildcard(_dataAccess);
var _reducerHandler = require('./reducerHandler');var _reducerHandler2 = _interopRequireDefault(_reducerHandler);
var _decodeHtmlStr = require('./decodeHtmlStr');var _decodeHtmlStr2 = _interopRequireDefault(_decodeHtmlStr);
var _momentHelper = require('./momentHelper');var momentHelper = _interopRequireWildcard(_momentHelper);
var _cls = require('./cls');var _cls2 = _interopRequireDefault(_cls);
var _filterBy = require('./filterBy');var _filterBy2 = _interopRequireDefault(_filterBy);
var _listenKeyDown = require('./listenKeyDown');var _listenKeyDown2 = _interopRequireDefault(_listenKeyDown);
var _filterValidCustomProps = require('./filterValidCustomProps');var _filterValidCustomProps2 = _interopRequireDefault(_filterValidCustomProps);
var _browser = require('./browser');var _browser2 = _interopRequireDefault(_browser);
var _shallowEqual = require('./shallowEqual');var _shallowEqual2 = _interopRequireDefault(_shallowEqual);function _interopRequireWildcard(obj) {if (obj && obj.__esModule) {return obj;} else {var newObj = {};if (obj != null) {for (var key in obj) {if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key];}}newObj.default = obj;return newObj;}}function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.


toQueryString = _toQueryString2.default;exports.
isPromise = _isPromise2.default;exports.
isFSA = _isFSA2.default;exports.
createFSA = _createFSA2.default;exports.
createReducer = _createReducer2.default;exports.
normalizeKeys = _normalizeKeys2.default;exports.
DataAccess = DataAccess;exports.
reducerHandler = _reducerHandler2.default;exports.
decodeHtmlStr = _decodeHtmlStr2.default;exports.
momentHelper = momentHelper;exports.
cls = _cls2.default;exports.
filterBy = _filterBy2.default;exports.
listenKeyDown = _listenKeyDown2.default;exports.
filterValidCustomProps = _filterValidCustomProps2.default;exports.
browser = _browser2.default;exports.
shallowEqual = _shallowEqual2.default;