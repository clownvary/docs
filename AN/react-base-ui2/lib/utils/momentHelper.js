'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _moment = require('moment');var _moment2 = _interopRequireDefault(_moment);
var _isString = require('lodash/isString');var _isString2 = _interopRequireDefault(_isString);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

var isSame = function isSame(momentA, momentB) {
  if (!momentA && !momentB) {
    return true;
  }

  if (momentA && !momentB || !momentA && momentB) {
    return false;
  }

  var a = (0, _moment2.default)(momentA);
  var b = (0, _moment2.default)(momentB);
  return a.isSame(b);
};

var isInRange = function isInRange(momentValue, momentMin, momentMax) {
  var value = momentValue ? (0, _moment2.default)(momentValue) : null;

  if (!value || !value.isValid()) return false;

  var min = momentMin ? (0, _moment2.default)(momentMin) : null;
  if (min && min.isValid()) {
    if (value.isBefore(min)) {
      return false;
    }
  }

  var max = momentMax ? (0, _moment2.default)(momentMax) : null;
  if (max && max.isValid()) {
    if (value.isAfter(max)) {
      return false;
    }
  }

  return true;
};

var isValid = function isValid(momentValue) {
  var value = momentValue ? (0, _moment2.default)(momentValue) : null;
  return value && value.isValid();
};

var isSameTime = function isSameTime(timeA, timeB) {var format = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 'HH:mm:ss';
  var defaultYear = '1970-01-01';
  var defaultYearFormat = 'YYYY-MM-DD';
  var defaultTimeFormat = 'HH:mm:ss';

  if (_moment2.default.isMoment(timeA) && _moment2.default.isMoment(timeB)) {
    return timeA.format(defaultTimeFormat) === timeB.format(defaultTimeFormat);
  }

  if ((0, _isString2.default)(timeA) && (0, _isString2.default)(timeB)) {
    if (timeA === timeB) {
      return true;
    }

    var momentTimeA = (0, _moment2.default)(defaultYear + ' ' + timeA, defaultYearFormat + ' ' + format);
    var momentTimeB = (0, _moment2.default)(defaultYear + ' ' + timeB, defaultYearFormat + ' ' + format);

    return momentTimeA.format(defaultTimeFormat) === momentTimeB.format(defaultTimeFormat);
  }

  return false;
};

var createMoment = function createMoment(value) {
  if (value) {
    var val = (0, _moment2.default)(value);

    if (val.isValid()) {
      return val;
    }
  }

  return null;
};exports.default =

{
  isSame: isSame,
  isInRange: isInRange,
  isValid: isValid,
  createMoment: createMoment,
  isSameTime: isSameTime };module.exports = exports['default'];