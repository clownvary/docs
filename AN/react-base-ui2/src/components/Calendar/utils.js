import moment from 'moment';
import isArray from 'lodash/isArray';
import uniqWith from 'lodash/uniqWith';

const getTenYearRange = (mmomentDate) => {
  const year = mmomentDate.year();
  const begin = year - (year % 10);
  const end = begin + 9;
  return `${begin}-${end}`;
};

const getWeeks = (currentDate, firstDayOfWeek) => {
  const now = currentDate.clone();
  const veryShortWeekdays = [];
  const weekDays = [];
  const localeData = currentDate.localeData();
  for (let dateColIndex = 0; dateColIndex < 7; dateColIndex += 1) {
    const index = (firstDayOfWeek + dateColIndex) % 7;
    now.day(index);
    weekDays[dateColIndex] = localeData.weekdays(now);
    veryShortWeekdays[dateColIndex] = weekDays[dateColIndex].substring(0, 1);
  }

  return { veryShortWeekdays, weekDays };
};

const compareByFormat = (date1, date2, format) =>
  date1.format(format) === date2.format(format);

const getSafeValue = (value) => {
  if (!isArray(value)) {
    value = [value];
  }
  const safeValue = value.map(v => v && moment(v)).filter(v => v && v.isValid());

  return uniqWith(safeValue, (m1, m2) => m1.isSame(m2, 'day'));
};

const sortedDates = value => (value.map(d => d.clone())
  .sort((a, b) => {
    if (a.isAfter(b, 'day')) {
      return 1;
    }
    if (a.isBefore(b, 'day')) {
      return -1;
    }
    return 0;
  }));

export default {
  getTenYearRange,
  getWeeks,
  compareByFormat,
  getSafeValue,
  sortedDates
};
