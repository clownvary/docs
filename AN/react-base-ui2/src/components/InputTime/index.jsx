import InputTime, { InputTimeProps } from './InputTime';

export {
  InputTime,
  InputTimeProps
};

export default InputTime;
