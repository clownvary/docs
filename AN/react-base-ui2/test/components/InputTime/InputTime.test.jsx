import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import InputTime from '../../../src/components/InputTime';
import { createMoment } from '../../../src/utils/momentHelper';

const setup = (props = {}) => mount(<InputTime {...props} />);

describe('components/InputDate/InputTime.jsx', () => {
  let wrapper;
  const m1 = createMoment('2018-01-01T00:00:00+00:00').utc();

  beforeEach(() => {
    const value = m1;
    wrapper = setup({ value });
  });

  test('rendering', () => {
    const instance = wrapper.instance();
    expect(instance.isBlank()).toBe(false);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
