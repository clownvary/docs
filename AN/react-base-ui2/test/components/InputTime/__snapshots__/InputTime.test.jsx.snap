// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/InputDate/InputTime.jsx rendering 1`] = `
<InputTime
  allowBlank={true}
  allowKeySpin={true}
  allowMouseSpin={false}
  ariaLabel="Input box"
  className=""
  currency="USD"
  data-qa-id=""
  disabled={false}
  flexibleCalendar={false}
  format="t"
  icon=""
  iconHint=""
  id=""
  items={Array []}
  listClassName=""
  listConfig={Object {}}
  listMaxHeight="300px"
  listPopupOptions={Object {}}
  listWidth="auto"
  name=""
  onBlur={[Function]}
  onCalendarClose={[Function]}
  onCalendarOpen={[Function]}
  onFocus={[Function]}
  onListClose={[Function]}
  onListOpen={[Function]}
  onTextChange={[Function]}
  onValueChange={[Function]}
  placeHolder=""
  prefix="an"
  showClear={false}
  showSpinner={false}
  showTrigger={false}
  showTrigger2={true}
  size="md"
  textAlign="left"
  timeStep={30}
  triggerIcon="icon-calendar-m"
  triggerIcon2="icon-clock-m"
  triggerIconHint=""
  triggerIconHint2=""
  triggerIconToggle=""
  triggerIconToggle2=""
  value={"2018-01-01T00:00:00.000Z"}
>
  <div
    className="an-input-time input-group buttons-1 input-group--md"
  >
    <input
      aria-label="Input box"
      className="input input-group__field textalign-left input__field"
      data-qa-id=""
      disabled={false}
      id=""
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onDragEnter={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      onMouseUp={[Function]}
      onPaste={[Function]}
      placeholder=""
      type="text"
    />
    <a
      aria-label=""
      className="input-group__item button-toggler"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <i
        aria-label="chevron icon"
        className="icon icon-clock-m"
      />
    </a>
  </div>
</InputTime>
`;
